tests/e2e/e2e_test.go

```diff
@@ -2116,6 +2116,19 @@ func TestE2E(t *testing.T) {
 					E2ELog(t, "Tainting node failed")
 					return err
 				}
+
+				removeTaintClosure := func() {
+					_ = wait.Poll(retryInterval, timeout, func() (bool, error) {
+						removeTaintErr := removeNodeTaint(t, f, taintedNode.Name, taintKey)
```

Comment:
why not add the retry in the `removeNodeTaint` function directly? That way this retry will happen every time a node taint removal is triggered (which is an operation that may fail).

Reply:
no reason except I wanted to keep the function simple. But we do have retries in other helper functions, so it's probably fine.